pkg/sql/opt/optbuilder/plpgsql.go

```diff
@@ -1055,6 +1055,9 @@ func (b *plpgsqlBuilder) buildPLpgSQLStatements(stmts []ast.Statement, s *scope)
 				col.scalar = b.ob.buildRoutine(proc, def, callCon.s, callScope, b.colRefs)
 			})
 			b.ob.constructProjectForScope(callCon.s, callScope)
+			if overload.Volatility == volatility.Volatile {
```

Comment:
I'm wondering if this should be `!= volatility.Leakproof` instead? Suppose the called SP could divide by zero if a certain argument is zero. It seems like optimizing out the call would still change the behavior of the query.